src/VisualStudio/Core/Def/Implementation/ProjectSystem/LinkedFileUtilities.cs

```diff
@@ -70,9 +70,7 @@ private IVsHierarchy GetSharedItemContextHierarchy(IVisualStudioHostDocument doc
                 return null;
             }
 
-            var sharedHierarchy = document.SharedHierarchy;
```

Comment:
@Pilchie yes, we need to figure out when we cache the info. unfortunately we can't remove this due to supporting filtering in error list for project k. we can't ask the question from background thread using IVsHierarchy.

I made not to use the cache for existing code but use it only in error list. for RTM, we should figure out right way to cache this.